native/src/main/java/io/ballerina/stdlib/xslt/XsltTransformer.java

```diff
@@ -115,12 +120,13 @@ public static Object transform(BXml xmlInput, BXml xslInput, BMap<BString, Objec
      * @param transformer The transformer to be used
      * @param paramInput The input parameter map
      */
-    private static void applyParameters(Transformer transformer, BMap<BString, Object> paramInput) {
+    private static void applyParameters(Transformer transformer,
+                                        BMap<BString, Object> paramInput) throws Exception {
         BIterator<?> iterator = paramInput.getIterator();
         while (iterator.hasNext()) {
             BArray next = (BArray) iterator.next();
             BString key = (BString) next.get(0);
-            Object value = next.get(1);
+            Object value = (next.get(1) instanceof BXmlItem) ? convertToDocument(next.get(1).toString()) : next.get(1);
```

Comment:
Will this work as intended if I pass in lets say `table<map<int[]>>` (which is `anydata`). Perhaps it is better to restrict the signature of the transform function to allow only `map<string|decimal|xml>`.WDYT?